packages/app/obojobo-document-engine/src/scripts/viewer/components/viewer-app.js

```diff
@@ -350,7 +359,21 @@ export default class ViewerApp extends React.Component {
 
 			delete this.leaveEvent
 			delete this.leftEpoch
+			return
 		}
+
+		// first view
+		this.sendViewStart()
```

Comment:
It's probably fine, but I'd feel a little bit better using a flag. Something either like `this.isViewed` or `this.state.isViewed`, maybe `this.hasSentInitialView`? - Then inside `sendViewStart` it could set the flag to true and then no longer allow the event to be sent once true.

Reply:
I changed the name some, it still doesn't conform to this suggestion, but it's better than what it was.

Reply:
I'm not seeing something, but it seems to be that sendViewStart will get called every-time onVisibilityChange gets called (although that doesn't actually appear to be happening). Also thinking we should change to `sendInitialViewEvent` so it doesn't get confused with the similarly named visit:start event

Reply:
return statements are easy to miss above